apps/glusterfs/app_node.go

```diff
@@ -320,6 +321,49 @@ func (a *App) NodeDelete(w http.ResponseWriter, r *http.Request) {
 				return err
 			}
 
+			volumes, err := ListCompleteVolumes(tx)
+			if err != nil {
+				logger.Err(err)
+				return err
+			}
+			for _, volID := range volumes {
+				volEntry, err := NewVolumeEntryFromId(tx, volID)
+				if err != nil {
+					logger.LogError("Get volume entry for ID %s Failed with error %s", volID, err.Error())
+					continue
+				}
+				if volEntry.Info.Block {
+					for _, id := range volEntry.Info.BlockInfo.BlockVolumes {
+
+						blockEntry, err := NewBlockVolumeEntryFromId(tx, id)
+						if err != nil {
+							logger.LogError("Get block volume entry for ID %s Failed with error %s", id, err.Error())
+							continue
+						}
+
+						if err = blockEntry.updateHosts(txdb); err != nil {
+							logger.LogError("Update block hosts for ID %s Failed with error %s", id, err.Error())
+							continue
+						}
+
+						if err := blockEntry.Save(tx); err != nil {
+							logger.LogError("Save block entry for ID %s Failed with error %s", id, err.Error())
+							continue
+						}
+					}
+				}
+				//update mountpoint info
+				if err = volEntry.updateMountInfo(txdb); err != nil {
```

Comment:
I don't think this function will do what we want. Based on my reading of the code this may wipe out any other options in the map other than backup-volfile-servers.

Additionally, this function has the same problem as I mentioned above about repeatedly going to the db only to get the same results over and over again in a loop.

I suggest:
* Extracting (as you've already partially done) the inner parts of `updateMountInfo` into new functions:
  * One that iterates over the nodes in the cluster and returns a list of the storage host names of those nodes. This function would be standalone and not attached to any struct.
  * Make the latter half of updateMountInfo into a helper function that takes a list of hosts and sets/updates the hosts and Options.
* Call it from updateMountInfo and before the start of the loop over the volumes.
* Now the loop over the block volumes sets .Hosts to the new list
* The loop over the volumes uses the helper function to update (& not overwrite) the mountpoint and options.